bpf/process/types/basic.h

```diff
@@ -989,7 +968,7 @@ selector_arg_offset(__u8 *f, struct msg_generic_kprobe *e, __u32 selector)
 	}
 
 	/* Making binary selectors fixes size helps on some kernels */
-	asm volatile("%[seloff] &= 0xeff;\n" ::[seloff] "+r"(seloff) :);
+	asm volatile("%[seloff] &= 0xfff;\n" ::[seloff] "+r"(seloff) :);
```

Comment:
nice, I checked and we use `val &= 0xeff` on other places, it's probably all wrong ;-)

Reply:
Ah good point. Added a separate patch to fix this. Let's see what happens 🤞🏼.

Reply:
Hmm... verification seems to be failing with this patch, so I removed it for now.

```diff
```


```
Verifying bpf/objs/bpf_generic_kprobe_v53.o... Failed!
regs=40 stack=0 before 194: (57) r6 &= 4095
regs=40 stack=0 before 193: (15) if r0 == 0x0 goto pc+1924
regs=40 stack=0 before 192: (85) call bpf_map_lookup_elem#1
regs=40 stack=0 before 190: (18) r1 = 0xffff9867ee4da000
regs=40 stack=0 before 189: (07) r2 += -308
regs=40 stack=0 before 188: (bf) r2 = r10
regs=40 stack=0 before 187: (c5) if r6 s< 0x2 goto pc+1930
 R0=map_value(id=0,off=0,ks=4,vs=112,imm=0) R1_w=inv(id=0) R2_w=inv(id=0,umax_value=4,var_off=(0x0; 0x7)) R3_w=inv(id=0,umax_value=255,var_off=(0x0; 0xff)) R6_rw=invP(id=0,umax_value=1047,var_off=(0x0; 0x7ff)) R7_w=map_value(id=0,off=0,ks=4,vs=4096,umax_value=1047,var_off=(0x0; 0x7ff),s32_max_value=2047,u32_max_value=2047) R8=map_value(id=0,off=0,ks=4,vs=4096,umax_value=1023,var_off=(0x0; 0x3ff)) R9=inv(id=10) R10=fp0 fp-8=????mmmm fp-16=????mmmm fp-40=????mmmm fp-304=mmmmmmmm fp-312_r=mmmm???? fp-320=map_value fp-328=ctx
parent already had regs=40 stack=0 marks
; int action = actions->act[i];
215: (61) r9 = *(u32 *)(r0 +4)
 R0_w=map_value(id=0,off=0,ks=4,vs=4096,umax_value=4095,var_off=(0x0; 0xfff)) R1_w=inv(id=0) R2_w=invP(id=0,umax_value=4095,var_off=(0x0; 0xfff)) R3_w=inv(id=0,umax_value=255,var_off=(0x0; 0xff)) R6_w=invP(id=17,umax_value=1047,var_off=(0x0; 0x7ff)) R7=map_value(id=0,off=0,ks=4,vs=4096,umax_value=1047,var_off=(0x0; 0x7ff),s32_max_value=2047,u32_max_value=2047) R8=map_value(id=0,off=0,ks=4,vs=4096,umax_value=1023,var_off=(0x0; 0x3ff)) R9_w=inv(id=10) R10=fp0 fp-8=????mmmm fp-16=????mmmm fp-40=????mmmm fp-304=mmmmmmmm fp-312=mmmm???? fp-320=map_value fp-328=ctx
invalid access to map value, value_size=4096 off=4099 size=4
R0 max value is outside of the allowed memory range
processed 241 insns (limit 1000000) max_states_per_insn 0 total_states 20 peak_states 20 mark_read 16
```

Reply:
made an issue: https://github.com/cilium/tetragon/pull/458#discussion_r985674185 to follow up.